code/datums/ai/controllers/troll.dm

```diff
@@ -20,3 +20,70 @@
 	)
 
 	idle_behavior = /datum/idle_behavior/idle_random_walk
+
+/datum/ai_controller/bog_troll
+	movement_delay = 0.7 SECONDS
+
+	ai_movement = /datum/ai_movement/hybrid_pathing
+
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/allow_items()
+
+	)
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target/bog_troll,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree/bog_troll,
+
+		/datum/ai_planning_subtree/find_dead_bodies/bog_troll,
+		/datum/ai_planning_subtree/eat_dead_body/bog_troll,
+		/datum/ai_planning_subtree/no_target_hide,
+	)
+
+	idle_behavior = /datum/idle_behavior/nothing
+
+/datum/ai_controller/troll/cave
+	movement_delay = 0.7 SECONDS
+
+	ai_movement = /datum/ai_movement/hybrid_pathing
+
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/allow_items()
+
+	)
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/flee_target,
+
+		/datum/ai_planning_subtree/simple_find_target/troll,
+		/datum/ai_planning_subtree/find_food/troll,
+		/datum/ai_planning_subtree/targeted_mob_ability,
```

Comment:
wanna make this cooler? create a new flag like BB_TROLL_ROCK, and make this its own type so targetted_mob_ability/rock_throw, then check if it has a troll_rock, if not set a find behavior to a rock or make them scavenge. Then if they have a rock throw it.